### PR TITLE
Do not let promoted constants mess up the previous summary

### PR DESCRIPTION
**Pull request by hermanventer**
## Description

Looking into why the outer fixed point does not converge I found two minor problems that are fixed in this PR:
1) The default equality comparison for summaries compared the analysis time field, which clearly is neither deterministic nor relevant to the fixed point computation.
2) When a function body has embedded bodies for promoted constants, these bodies should not be summarized. Not only were they summarized, but the summaries were written to the cache, which made the before and after comparisons useless because the current (corrupted) version of the cache was used as the before version.

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] API change with a documentation update
- [ ] Additional test coverage

## How Has This Been Tested?
cargo test; validate.sh

### CODE REVIEW

File: checker/src/summaries.rs [hunk 1]
```diff
@@ -94,6 +94,30 @@ pub struct Summary {
     pub unwind_side_effects: Vec<(Path, AbstractValue)>,
 }
 
+impl PartialEq for Summary {
```
Comment (zlandau) [inline checker/src/summaries.rs]: Not a fan of a single big return a == b && this == that and so on, eh?

File: checker/src/visitors.rs [hunk 2]
```diff
@@ -535,6 +536,28 @@ impl<'a, 'b: 'a, 'tcx: 'b, E> MirVisitor<'a, 'b, 'tcx, E> {
...
 
+    /// Computes a fixed point over the blocks of the MIR for a promoted constant block
+    fn visit_promoted_constants_block(&mut self, function_name: &str) {
+        let mut block_indices = self.get_sorted_block_indices();
+
+        let (mut in_state, mut out_state) =
+            <MirVisitor<'a, 'b, 'tcx, E>>::initialize_state_maps(&block_indices);
+
+        // The entry block has no predecessors and its initial state is the function parameters
+        // which we omit here so that we can lazily provision them with additional context.
```
Comment (zlandau) [inline checker/src/visitors.rs]: A bit of a nit, but since in general you make your explanations fairly descriptive: I found this a bit hard to parse, maybe just some parenthesis (assuming I even did manage to parse it correctly):

```
// The entry block has no predecessors and its initial state is the function parameters
// (which we omit here so that we can lazily provision them with additional context).
```

File: checker/src/summaries.rs (hunk 1 above)
Comment (hermanventer) [reply checker/src/summaries.rs]: It really helps with debugging when you can set break points at particular parts of the expression.
